### PR TITLE
Add props to Carousel component

### PR DESCRIPTION
**Pull request by craigpalermo**

### CODE REVIEW

File: packages/carousel/src/layoutToFlexWidths.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export const layoutToFlexWidths = (layout, totalSlides) => {
```
Comment (craigpalermo) [inline packages/carousel/src/layoutToFlexWidths.js:1]: - [ ] unit tests

File: packages/carousel/src/layoutToFlexWidths.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    return []
+  }
+
+  const layoutValues = layout?.split('-').map((i) => parseInt(i))
+  const numLayoutValues = layoutValues.length
+
+  let widths = []
+
+  for (let i = 0; i < totalSlides; i++) {
+    if (i < layoutValues.length) {
```
Comment (craigpalermo) [inline packages/carousel/src/layoutToFlexWidths.js:12]: Optimize by iterating over `layoutValues` instead

File: packages/carousel/src/Carousel.js [hunk 3]
```diff
@@ -53,7 +66,41 @@ export const Carousel = ({
...
+   * into pages with the same number of items as the "layout" page. Not compatible with vertical orientation. */
+  layout: PropTypes.oneOf([
+    '50-50',
+    '25-50-25',
+    '75-25',
+    '25-75',
+    '33-33-33',
+    '66-33',
+    '33-66',
+    '25-25-25-25',
```
Comment (craigpalermo) [inline packages/carousel/src/Carousel.js:87]: - [ ] verify that these are the correct layouts to support

File: packages/carousel/src/Carousel.js [hunk 4]
```diff
@@ -1,39 +1,56 @@
...
 import { v4 as uuidv4 } from 'uuid'
 
+import { layoutToFlexWidths } from './layoutToFlexWidths'
 import { CarouselWrapper } from './Carousel.styles'
 
 const getSlideKey = moize(uuidv4)
 
 export const Carousel = ({
   children,
+  showDots = false,
```
Comment (ryan-hunt-priceline) [inline packages/carousel/src/Carousel.js:14]: NB: should we define the defaults with `defaultProps` property?

File: packages/carousel/src/Carousel.js [hunk 5]
```diff
@@ -53,7 +70,34 @@ export const Carousel = ({
...
+   * this will be overridden by the number of items in the layout.
+   * E.g. `layout={'25-50-25'}` will result in 3 slides shown per page. */
   visibleSlides: PropTypes.number,
+  /** Show clickable navigation dots */
   showDots: PropTypes.bool,
+  /** Show forward/back navigation buttons */
   showForwardBackBtns: PropTypes.bool,
+  /** Percent width for each of the first N slides to use as a layout. Remaining slides will be divided
+   * into pages with the same number of items as the "layout" page. Not compatible with vertical orientation. */
+  layout: PropTypes.oneOf(['25-25-25-25', '33-33-33', '50-50', '40-60', '60-40', '25-75', '75-25']),
```
Comment (ryan-hunt-priceline) [inline packages/carousel/src/Carousel.js:83]: Should it be `60-40` or do we want it to be `66-33`?

Comment (craigpalermo) [reply packages/carousel/src/Carousel.js:83]: I got these values from the Contentful model that Marketing will be using. @wesleymartin85 can you confirm these values are correct?

File: packages/carousel/src/Carousel.js (hunk 4 above)
Comment (sdalonzo) [reply packages/carousel/src/Carousel.js:14]: perhaps - defaultProps will render in storybook docs i think

**Review by sdalonzo — APPROVED** [review]

**Pull request merged**